src/valuetype_idtype.ts

```diff
@@ -55,9 +56,17 @@ function guessIDType(def: ITypeDefinition, data: any[], accessor: (row: any) =>
   return def;
 }
 
-function isIDType(name: string, index: number, data: any[], accessor: (row: any) => string, sampleSize: number) {
-  //TODO guess the first one is it most of the times
-  return index === 0 ? 0.5 : 0;
+async function isIDType(name: string, index: number, data: any[], accessor: (row: any) => string, sampleSize: number) {
+  const pluginPromises: Promise<number>[] = [];
+  plugins.list('idTypeDetector').forEach((pluginDesc) => {
+    pluginPromises.push(pluginDesc.load().then((factory) => {
+      const plugin = factory.factory();
+      return plugin.detectIDType(data, accessor, sampleSize);
+    }));
+  });
+
+  const values = await Promise.all(pluginPromises);
+  return Math.max(...values) / 2;
```

Comment:
it's a relic of an error I had in which the wrong type was suggested and I was trying something. So it was committed by accident and is already removed in the new code